ConfigSpace/util.pyx

```diff
@@ -436,3 +436,185 @@ def fix_types(configuration: dict,
             else:
                 raise TypeError("Unknown hyperparameter type %s" % type(param))
     return configuration
+
+cimport cython
+@cython.boundscheck(True)  # Activate bounds checking
+@cython.wraparound(True)  # Activate negative indexing
+def generate_grid(configuration_space: ConfigurationSpace,
+                 num_steps_dict: Union[None,  Dict[str, int]] = None,
+                 ) -> List[Configuration]:
+    """
+    Generates a grid of Configurations for a given ConfigurationSpace. Can be used, for example, for grid search.
+
+    Parameters
+    ----------
+    configuration_space: :class:`~ConfigSpace.configuration_space.ConfigurationSpace`
+        The Configuration space over which to create a grid of HyperParameter Configuration values. It knows the types for all parameter values.
+
+    num_steps_dict: dict
+        A dict containing the number of points to divide the grid side formed by Hyperparameters which are either of type UniformFloatHyperparameter or type UniformIntegerHyperparameter. The keys in the dict should be the names of the corresponding Hyperparameters and the values should be the number of points to divide the grid side formed by the corresponding Hyperparameter in to.
+
+    Returns
+    -------
+    list
+        List containing Configurations. It is a cartesian product of tuples of HyperParameter values. Each tuple lists the possible values taken by the corresponding HyperParameter. Within the cartesian product, in each element, the ordering of HyperParameters is the same for the OrderedDict within the ConfigurationSpace.
+    """
+
+    def get_value_set(num_steps_dict, hp_name):

```

Comment:
Could you please add type annotations here, too?

Reply:
Done